.github/scripts/check_formalities.sh

```diff
@@ -48,19 +52,63 @@ for commit in $(git rev-list HEAD ^origin/"$BRANCH"); do
 		RET=1
 	fi
 
+	echo
+	info "Checking subject:"
 	subject="$(git show -s --format=%s "$commit")"
-	if echo "$subject" | grep -q -e '^[0-9A-Za-z,+/_-]\+: ' -e '^Revert '; then
-		success "Commit subject line seems OK ($subject)"
-	elif echo "$subject" | grep -iq '^Translated using Weblate.*'; then
-		warn "Weblate commit subject line exception: $subject"
-	elif echo "$subject" | grep -iq '^Added translation using Weblate.*'; then
-		warn "Weblate commit subject line exception: $subject"
+	echo "$subject"
+
+	# Check subject format
+	if echo "$subject" | grep -iq -e '^Translated using Weblate.*' -e '^Added translation using Weblate.*'; then
+		warn "Weblate commit subject line exception"
+	elif echo "$subject" | grep -q -e '^[0-9A-Za-z,+/_-]\+: [a-z]' -e '^Revert '; then
+		success "Commit subject line format seems OK"
+	elif echo "$subject" | grep -q -e '^[0-9A-Za-z,+/_-]\+: [A-Z]'; then
+		err "First word after prefix in subject should not be capitalized"
+		RET=1
+	elif echo "$subject" | grep -q -e '^[0-9A-Za-z,+/_-]\+: '; then
+		# Handles cases when there's a prefix but the check for capitalization fails (e.g. no word
+		# after prefix)
+		err "Commit subject line MUST start with '<package name>: ' and be followed by a lower-case word"
+		RET=1
 	else
-		err "Commit subject line MUST start with '<package name>: ' ($subject)"
+		err "Commit subject line MUST start with '<package name>: '"
+		RET=1
+	fi
+
+	if echo "$subject" | grep -q '\.$'; then
+		err "Commit subject line should not end with a period"
 		RET=1
 	fi
 
+	# Check subject length
+	if [ ${#subject} -gt "$MAX_SUBJECT_LEN" ]; then
+		warn "Commit subject line is longer than $MAX_SUBJECT_LEN characters (is ${#subject})"
```

Comment:
I understand that the more rules there are, the more complicated it will simply be, but at this point I think that enforcing a rule that the commit subject should be brief, precise, simple and concise is something we want, and that it shouldn't be a warning, but an error.
An example is this pull request: https://github.com/openwrt/packages/pull/27564, the GitHub UI hid that subject because it's simply too long, which should have helped tell the user that it's long, but apparently that didn't happen. Just to be sure, here's a link to the commit itself: https://github.com/openwrt/packages/pull/27564/commits/a5a7ca783e5f383824fc2b69c1a336dad17b158e

On the other hand, the commit description can be a warning. Wrapping or not wrapping at 75 characters per line doesn't bother me personally.

Reply:
Hm... Sounds great!